libusb/os/windows_winusb.c

```diff
@@ -722,10 +723,50 @@ static void winusb_exit(struct libusb_context *ctx)
 	exit_dlls();
 }
 
+static BOOL device_io_control(HANDLE event, HANDLE hDevice, DWORD dwIoControlCode, LPVOID lpInBuffer, DWORD nInBufferSize, LPVOID lpOutBuffer, DWORD nOutBufferSize, LPDWORD lpBytesReturned)
+{
+	OVERLAPPED overlapped;
+	DWORD error;
+	BOOL ret = TRUE;
+
+	ResetEvent(event);
+	SecureZeroMemory(&overlapped, sizeof(overlapped));
+	overlapped.hEvent = event;
+
+	if (DeviceIoControl(hDevice, dwIoControlCode, lpInBuffer, nInBufferSize, lpOutBuffer, nOutBufferSize, lpBytesReturned, &overlapped)) goto end;
+
+	error = GetLastError();
+	if (error != ERROR_IO_PENDING) {
+		CancelIo(hDevice);
+		ret = FALSE;
+		goto end;
+	}
+
+	DWORD wret = WaitForSingleObject(event, IO_WAIT_TIME);
+	if (wret != WAIT_OBJECT_0) {
+		CancelIo(hDevice);
+		ret = FALSE;
+		goto end;
+	}
+	else {
+		DWORD bytes_written = 0;
+		ret = GetOverlappedResult(hDevice, &overlapped, &bytes_written, TRUE);
+		assert(bytes_written == *lpBytesReturned);
+
+		if (!ret) {
+			CancelIo(hDevice);
```

Comment:
if we're here - `WaitForSingleObject` has returned the `WAIT_OBJECT_0`, which means there is no pending operations this function has started
probably best not to interfere with other IO operations that might be happening on the device
```suggestion
```